geode-core/src/main/java/org/apache/geode/cache/configuration/CacheConfig.java

```diff
@@ -1024,11 +1025,22 @@ public void setVersion(String value) {
     this.version = value;
   }
 
+  // this supports looking for sub regions
   public RegionConfig findRegionConfiguration(String regionPath) {
     if (regionPath.startsWith(SEPARATOR)) {
       regionPath = regionPath.substring(1);
     }
-    return find(getRegions(), regionPath);
+    List<RegionConfig> regions = getRegions();
+    RegionConfig found = null;
+    for (String regionToken : regionPath.split(SEPARATOR)) {
+      found = Identifiable.find(regions, regionToken);
```

Comment:
I feel this logic need some comments or use better name for variables found and regions. 

It tried to find the last layer of subregion's RegionConfig and validate it exists.